File: playlist.java
import java.io.*;
import java.util.*;

public class playlist {
    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine()), ans = 0, start = -1;
        StringTokenizer st = new StringTokenizer(br.readLine());
        HashMap<Integer, Integer> map = new HashMap<>(); 
        for (int i = 0; i < n; i++) {
            int k = Integer.parseInt(st.nextToken());
            
            if (map.containsKey(k)) {
                start = Math.max(start, map.get(k));
            }
            ans = Math.max(ans, i - start);
            map.put(k, i);
        }
        System.out.println(ans);
    }
}

File: playlist.cpp
#include <iostream>
#include <unordered_map>
#include <algorithm>

using namespace std;

int main(void) {
    int n, ans = 0;
    cin >> n;

    unordered_map<int, int> map;
    map.reserve(1024);
    int start = -1;
    for (int i = 0; i < n; i++) {
        int k;
        cin >> k;
        if (map.contains(k)) {
            start = max(start, map[k]);
        }
        ans = max(ans, i - start);
        map[k] = i;
    }
    cout << ans << endl;
}
